algorithms/src/snark/marlin/ahp/indexer/constraint_system.rs

```diff
@@ -123,7 +141,18 @@ impl<F: Field> CS<F> for ConstraintSystem<F> {
         self.b.push(Self::make_row(&b(LinearCombination::zero())));
         self.c.push(Self::make_row(&c(LinearCombination::zero())));
 
+        self.mul_constraints.insert(self.num_constraints);
+        self.num_constraints += 1;
+    }
+
+    fn lookup(&mut self, _: &[LinearCombination<F>], table_index: usize) -> Result<Variable, SynthesisError> {
+        self.a.push(Self::make_row(&LinearCombination::zero()));
```

Comment:
Shouldn't the various `self.m.push(...)` be identical to `enforce` above?

Reply:
yep, good catch! seems i was just pushing zeros this whole time